In interpolate, use if instead of elif

dreiss opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37177 Update interpolate to use new upsample overloads
* #37176 Add interpolate-style overloads to aten::upsample* ops
* #37175 Add support for float[]? arguments in native_functions.yaml
* #37174 Add support for int[]? arguments in native_functions.yaml
* #37173 In interpolate, inline the call to _interp_output_size
* #37172 In interpolate, move exceptional cases to the bottom
* **#37171 In interpolate, use if instead of elif**



Every one of these branches returns or raises, so there's no need for elif.
This makes it a little easier to reorder and move conditions.

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992/)

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992)

dreiss committed:
In interpolate, use if instead of elif

Every one of these branches returns or raises, so there's no need for elif.
This makes it a little easier to reorder and move conditions.

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992/)

[ghstack-poisoned]

eellison reviewed (changes requested):
this is a reminder for me to do something that has been on my list but hasn't been pressing yet for any particular reason. 
If you have a function like:
```
@torch.jit.script
def test_foo(x: bool, y: bool):
    if x:
        return 1
    return 2
print(test_foo.code)
```
generates:
```
def test_foo(x: bool,
    y: bool) -> int:
  _0 = uninitialized(int)
  if x:
    _1, _2 = True, 1
  else:
    _1, _2 = False, _0
  if _1:
    _3 = _2
  else:
    _3 = 2
  return _3
```
while
```
@torch.jit.script
def test_foo(x: bool, y: bool):
    if x:
        return 1
    else:
        return 2
print(test_foo.code)
```
generates:
```
def test_foo(x: bool,
    y: bool) -> int:
  if x:
    _0 = 1
  else:
    _0 = 2
  return _0
```

The reason for this is how we destructure early returns into if statements. I can get around to adding a rewrite pass sometime next week, but before then we probably shouldn't add all of these extra if statements in a core function for a non-semantic change.

dreiss committed:
Update on "In interpolate, use if instead of elif"




Every one of these branches returns or raises, so there's no need for elif.
This makes it a little easier to reorder and move conditions.

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992/)

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992)

[ghstack-poisoned]

dreiss committed:
Update on "In interpolate, use if instead of elif"




Every one of these branches returns or raises, so there's no need for elif.
This makes it a little easier to reorder and move conditions.

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992/)

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992)

[ghstack-poisoned]

dreiss committed:
Update on "In interpolate, use if instead of elif"




Every one of these branches returns or raises, so there's no need for elif.
This makes it a little easier to reorder and move conditions.

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992/)

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992)

[ghstack-poisoned]

dreiss committed:
Update on "In interpolate, use if instead of elif"




Every one of these branches returns or raises, so there's no need for elif.
This makes it a little easier to reorder and move conditions.

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992/)

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992)

[ghstack-poisoned]

dreiss committed:
Update on "In interpolate, use if instead of elif"




Every one of these branches returns or raises, so there's no need for elif.
This makes it a little easier to reorder and move conditions.

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992/)

Differential Revision: [D21209992](https://our.internmc.facebook.com/intern/diff/D21209992)

[ghstack-poisoned]
